src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java

```diff
@@ -575,8 +575,8 @@ public KubernetesClient connect() throws KubernetesAuthException, IOException {
      */
     private boolean addProvisionedSlave(@Nonnull PodTemplate template, @CheckForNull Label label, int scheduledCount) throws Exception {
         int containerCap = getContainerCap();
-        if (containerCap == 0) {
-            return false;
+        if (containerCap == Integer.MAX_VALUE) { // don't check concurrency limits when set to "unlimited."
+            return true;
```

Comment:
https://github.com/jenkinsci/kubernetes-plugin/blob/23e40caa021679adc1de7f43eed7317cd094742f/src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java#L383-L384

Reply:
Or rather than calling the getter you could just check `containerCap == null`.

Reply:
In either case, `KubernetesCloudTest#testInstanceCap` fails because concurrency limit checking is skipped when the pod template has a limit but the overall cloud is unlimited.

FTR Pod templates are using `Integer.MAX_VALUE` to mean unlimited:
https://github.com/jenkinsci/kubernetes-plugin/blob/dedad44d243a94d7a4fe474340293dda5e3615ff/src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java#L306-L312

I'll toy around with it some more until tests pass.

Reply:
In version `1.27.0`, before all the changes to concurrency limits, setting the overall cloud to `0` skipped all concurrency limit checking, even if the pod template had one set. Is that the expected behavior?

I was thinking that the pod template level limit should be obeyed when the overall cloud is unlimited.

Reply:
Removed the `if` statement entirely in https://github.com/jenkinsci/kubernetes-plugin/pull/866/commits/464b64261c3bba254d51b009f228453e3a8130cc and things are looking better now cc @Vlatombe 

The behavior is also better now: if the global limit is "unlimited" then the pod template limit is still respected.